blocks/rich-text/index.js

```diff
@@ -829,15 +817,14 @@ export class RichText extends Component {
 			isSelected = false,
 			formatters,
 		} = this.props;
-		const { empty } = this.state;
 
 		const ariaProps = pickAriaProps( this.props );
 
 		// Generating a key that includes `tagName` ensures that if the tag
 		// changes, we unmount and destroy the previous TinyMCE element, then
 		// mount and initialize a new child element in its place.
 		const key = [ 'editor', Tagname ].join();
-		const isPlaceholderVisible = placeholder && ( ! isSelected || keepPlaceholderOnFocus ) && empty;
+		const isPlaceholderVisible = placeholder && ( ! isSelected || keepPlaceholderOnFocus ) && this.isEmpty;
```

Comment:
I'm thinking this change could potentially create an issue if the block is not rerendered when the `this.isEmpty` property changes. Do you think we should use `forceUpdate` when its value changes?

Reply:
Won't this component always rerender if the value props change as well?

Reply:
Yes, it should unless the parent component is "uncontrolled" or something. Granted we don't use it this way though

Reply:
Sounds good, but I don't want to go too deep into empty checks in this PR.